File: ex4/fifo.c
#include <stdio.h>
struct p{
	int time;
	char name;
};

void fifo(struct p ps[]){
	int now = 0;
	printf("here");
	for(int i = 0;i<sizeof(*ps);i++){
		printf("time spent:%d",now);
		printf("acting process:%s\n",ps[i].name);
	}
};

int main(void){
	struct p ps[5]={
		{5,'a'},{19,'b'},{3,'c'},{9,'d'},{2,'e'}
	};
	fifo(ps);
	return 0;
}

File: ex4/Fifo.java
public class Fifo{
    private static class P{
        int time;
        char pName;
        P(int time,char name){
            this.time = time;
            this.pName = name;
        }
    }

    public static void fifo(P[] ps){
        int time = 0;
        for(int i = 0 ; i < ps.length ; i++){
            System.out.println("Time:" + time + "  process:" + ps[i].pName);
            time += ps[i].time;
        }
        System.out.println("Time:" + time + "\nEnd");
    }

    public static void main(String[] args){
        P[] ps = {
            new P(1,'a'),
            new P(20,'b'),
            new P(3,'c'),
            new P(96,'d'),
            new P(2,'e')
        };
        fifo(ps);
    }
}